static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js

```diff
@@ -75,8 +75,13 @@ const getReaderRevenueRegion = (geolocation: string): ReaderRevenueRegion => {
 
 const getVisitCount = (): number => local.get('gu.alreadyVisited') || 0;
 
-const replaceCountryName = (text: string, countryName: string): string =>
-    text.replace(/%%COUNTRY_NAME%%/g, countryName);
+const replaceCountryName = (text: string, countryName: ?string): string => {
+    if (!countryName) {
+        if (text.includes('%%COUNTRY_NAME%%'))
+            throw new Error('Missing country name');
```

Comment:
👍

Reply:
Should we `catch` this error too? What will throwing this error produce on the client? I assume we haven't tested because we haven't reproduced.

Reply:
I've left that to the calling functions.
In the case of the epic, this is `buildEpicCopy`, which does catch the error and sets the `copy` to undefined, meaning it will fall back on the default epic copy.

In the case of the banner, it is caught [here](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/commercial/contributions-utilities.js#L789) and the default banner will be displayed (not a test banner)

Reply:
Sorry I was thinking of the `try`/`catch` pattern of error handling

Reply:
Got it